libs/api/windows_platform.cpp

```diff
@@ -13,83 +13,60 @@
 #include "windows_platform.hpp"
 #include "windows_platform_common.hpp"
 
-int
-create_map_internal(
-    uint32_t type,
-    uint32_t key_size,
-    uint32_t value_size,
-    uint32_t max_entries,
-    uint32_t inner_map_fd,
-    size_t section_offset,
-    ebpf_verifier_options_t options);
-
-static int
-create_map_windows(
-    uint32_t map_type,
-    uint32_t key_size,
-    uint32_t value_size,
-    uint32_t max_entries,
-    uint32_t inner_map_fd,
-    size_t section_offset,
-    ebpf_verifier_options_t options)
-{
-    int fd;
-    if (options.mock_map_fds) {
-        EbpfMapType type = get_map_type_windows(map_type);
-        fd = create_map_crab(type, key_size, value_size, max_entries, options);
-        cache_map_file_descriptor(map_type, key_size, value_size, max_entries, inner_map_fd, fd);
-        return fd;
-    }
-
-    return create_map_internal(map_type, key_size, value_size, max_entries, inner_map_fd, section_offset, options);
-}
-
 void
 parse_maps_section_windows(
-    std::vector<EbpfMapDescriptor>& map_descriptors,
+    std::vector<EbpfMapDescriptor>& verifier_map_descriptors,
     const char* data,
     size_t size,
     const struct ebpf_platform_t*,
-    ebpf_verifier_options_t options)
+    ebpf_verifier_options_t)
 {
-    if (size % sizeof(ebpf_map_definition_t) != 0) {
+    if (size % sizeof(ebpf_map_definition_in_file_t) != 0) {
         throw std::runtime_error(
             std::string("bad maps section size, must be a multiple of ") +
-            std::to_string(sizeof(ebpf_map_definition_t)));
+            std::to_string(sizeof(ebpf_map_definition_in_file_t)));
     }
 
-    auto mapdefs =
-        std::vector<ebpf_map_definition_t>((ebpf_map_definition_t*)data, (ebpf_map_definition_t*)(data + size));
+    auto mapdefs = std::vector<ebpf_map_definition_in_file_t>(
+        (ebpf_map_definition_in_file_t*)data, (ebpf_map_definition_in_file_t*)(data + size));
     for (int i = 0; i < mapdefs.size(); i++) {
         auto& s = mapdefs[i];
-        map_descriptors.emplace_back(EbpfMapDescriptor{
-            .original_fd = create_map_windows(
-                s.type,
-                s.key_size,
-                s.value_size,
-                s.max_entries,
-                s.inner_map_idx,
-                (i * sizeof(ebpf_map_definition_t)),
-                options),
+        uint32_t section_offset = (i * sizeof(ebpf_map_definition_in_file_t));
+
+        // The map file descriptors that appear in eBPF bytecode start at 1,
+        // in the order the maps appear in the maps section.
+        const int ORIGINAL_FD_OFFSET = 1;
```

Comment:
minor: We can move this outside the for loop.

Reply:
Done